### PR TITLE
Chapter 1.4

### LINKED ISSUE #28: Translate to Agda 1.4
This is mostly done (see #12, #19, #21), but #21 still needs merging and possibly improvement...?

### PR DESCRIPTION
**Pull request by williamdemeo**

### CODE REVIEW

**Comment by williamdemeo** [comment]
Just a few minor changes... plus solutions to the challenges.  @lemastero lmkwyt

File: src/content/1.4/kleisli-categories.tex [hunk 1]
```diff
@@ -3,8 +3,8 @@
 \lettrine[lhang=0.17]{Y}{ou've seen how to model} types and pure functions as a category. I also
 mentioned that there is a way to model side effects, or non-pure
 functions, in category theory. Let's have a look at one such example:
-functions that log or trace their execution. Something that, in an
-imperative language, would likely be implemented by mutating some global
+functions that log or trace their execution. In an imperative language,
+such functions would likely be implemented by mutating some global
```
Comment (lemastero) [inline src/content/1.4/kleisli-categories.tex:7]: Is this change intentional? We agree at some point to keep original wording of the book.

Comment (williamdemeo) [reply src/content/1.4/kleisli-categories.tex:7]: Yes, I thought the original wording needed improvement... but if you think we shouldn't mess with the original text, I can revert.  Lmkwyt.  Thanks.

**Review by lemastero — APPROVED** [review]

File: src/content/1.4/kleisli-categories.tex (hunk 1 above)
Comment (lemastero) [reply src/content/1.4/kleisli-categories.tex:7]: Let's keep the changes. Ultimately it's up to reviewers of the entire translation. As we want to make a huuuge PR at the end :)

File: src/content/1.4/code/agda/full.agda [hunk 2]
```diff
@@ -1,27 +1,32 @@
...
+_>=>_ : (a → Writer b) → (b → Writer c) → (a → Writer c)
 m1 >=> m2 = λ x →
   let (y , s1) = m1 x
       (z , s2) = m2 y
   in (z , s1 ++ s2)
 
+-- alternative --
+_≫=≫_ : (a → Writer b) → (b → Writer c) → a → Writer c
+m1 ≫=≫ m2 = λ x →
+  fst (m2 $ fst (m1 x)) , snd (m1 x) ++ snd (m2 $ fst (m1 x))
```
Comment (lemastero) [inline src/content/1.4/code/agda/full.agda]: We are computing `m1 x` three times here.

I think version with `where` is more idiomatic to Agda. At least it is often used in: https://www.cs.bham.ac.uk/~mhe/HoTT-UF-in-Agda-Lecture-Notes/

Comment (williamdemeo) [reply src/content/1.4/code/agda/full.agda]: Okay, I can remove that one.  I just thought it was worth seeing another way to present it.

As for computing `m1 x` three times, imho, when it comes to formalizing mathematics, computational efficiency should play no role... unless we are formalizing mathematics of computational efficiency! :)
